uplink/src/collector/telemetry.rs

```diff
@@ -0,0 +1,534 @@
+use flume::Sender;
+use log::error;
+use serde::Serialize;
+use sysinfo::{DiskExt, NetworkData, NetworkExt, ProcessExt, ProcessorExt, System, SystemExt};
+use tokio::time::Instant;
+
+use std::{
+    collections::HashMap,
+    sync::Arc,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
+
+use crate::base::{Buffer, Config, Package, Point, Stream};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Io error {0}")]
+    Io(#[from] std::io::Error),
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+struct LoadAvg {
+    /// Average load within one minute.
+    pub one: f64,
+    /// Average load within five minutes.
+    pub five: f64,
+    /// Average load within fifteen minutes.
+    pub fifteen: f64,
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+struct SysInfo {
+    kernel_version: String,
+    uptime: u64,
+    no_processes: usize,
+    #[serde(flatten)]
+    load_avg: LoadAvg,
+}
+
+impl SysInfo {
+    fn init(sys: &System) -> Self {
+        SysInfo {
+            kernel_version: match sys.kernel_version() {
+                Some(kv) => kv,
+                None => String::default(),
+            },
+            ..Default::default()
+        }
+    }
+
+    fn update(&mut self, sys: &System) {
+        self.uptime = sys.uptime();
+        self.no_processes = sys.processes().len();
+        let sysinfo::LoadAvg { one, five, fifteen } = sys.load_average();
+        self.load_avg = LoadAvg { one, five, fifteen };
+    }
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+struct Mem {
+    total_memory: u64,
+    available_memory: u64,
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+pub struct SystemStats {
+    sequence: u32,
+    timestamp: u64,
+    #[serde(flatten)]
```

Comment:
Can't we just use members of `Mem` and `SysInfo` here directly instead of creating new child structs?